src/util/bip32.rs

```diff
@@ -249,10 +401,10 @@ impl ExtendedPrivKey {
     pub fn derive_priv<C: secp256k1::Signing>(
         &self,
         secp: &Secp256k1<C>,
-        cnums: &[ChildNumber],
+        path: &DerivationPath,
```

Comment:
I'm ok with it either way. It adds some complexity but is probably worth it. I'm generally in favor of more generic code, but I recently experienced it from the user's point of view and saw how hard it can be to work with such code if there are no hints which types implement the needed traits. But here it's still trivial I think, maybe add documentation that hints at `DerivationPath`.